youtube_dl/extractor/tvplay.py

```diff
@@ -312,14 +312,19 @@ def _real_extract(self, url):
 
         # TODO: webvtt in m3u8
         subtitles = {}
-        sami_path = video.get('sami_path')
-        if sami_path:
-            lang = self._search_regex(
-                r'_([a-z]{2})\.xml', sami_path, 'lang',
-                default=compat_urlparse.urlparse(url).netloc.rsplit('.', 1)[-1])
-            subtitles[lang] = [{
-                'url': sami_path,
-            }]
+        sub_paths = [
+            video.get('sami_path'),
+            video.get('subtitles_webvtt'),
+            video.get('subtitles_for_hearing_impaired'),
+        ]
+        for path in sub_paths:
+            if path:
+                lang = self._search_regex(
+                    r'_(.*)(\.)', path, 'lang',
```

Comment:
Don't relax regexes.

Reply:
How would an optimal regex look like in this situation?
It should do the following:
`https://cdn-subtitles-mtgx-tv.akamaized.net/pitcher/20xxxxxx/2034xxxx/203476xx/20347612/5370102610/unified.subs.20347612_sv.vtt` **->** `sv`
`https://cdn-subtitles-mtgx-tv.akamaized.net/pitcher/20xxxxxx/2034xxxx/203476xx/20347612/5370102610/unified.subs.20347612_sv_sdh.vtt` **->** `sv_sdh`